### ISSUE TITLE
undefined method `[]' for nil:NilClass

### ISSUE THREAD

**Issue by eapache** [whitequark/parser#186]
A file containing the following:
```ruby
some_method do |*, token:, **|
end
```

Raises the exception in the title, with the following backtrace:
```
"/home/vagrant/.gem/ruby/2.1.5/gems/parser-2.2.0.2/lib/parser/builders/default.rb:1034:in `arg_name_collides?'",
 "/home/vagrant/.gem/ruby/2.1.5/gems/parser-2.2.0.2/lib/parser/builders/default.rb:1013:in `block in check_duplicate_args'",
 "/home/vagrant/.gem/ruby/2.1.5/gems/parser-2.2.0.2/lib/parser/builders/default.rb:1000:in `each'",
 "/home/vagrant/.gem/ruby/2.1.5/gems/parser-2.2.0.2/lib/parser/builders/default.rb:1000:in `check_duplicate_args'",
 "/home/vagrant/.gem/ruby/2.1.5/gems/parser-2.2.0.2/lib/parser/builders/default.rb:526:in `args'",
 "ruby21.y:1435:in `_reduce_382'",
 "/usr/lib/shopify-ruby/2.1.5-shopify1/lib/ruby/2.1.0/racc/parser.rb:258:in `_racc_do_parse_c'",
 "/usr/lib/shopify-ruby/2.1.5-shopify1/lib/ruby/2.1.0/racc/parser.rb:258:in `do_parse'",
 "/home/vagrant/.gem/ruby/2.1.5/gems/parser-2.2.0.2/lib/parser/base.rb:175:in `parse'",
 "/home/vagrant/.gem/ruby/2.1.5/gems/parser-2.2.0.2/lib/parser/base.rb:31:in `parse'",
 "/home/vagrant/.gem/ruby/2.1.5/gems/parser-2.2.0.2/lib/parser/base.rb:63:in `parse_file'",
```

**Comment by eapache** [comment]
I was parsing our entire code base and came across this pattern in our code base, then reduced it to the above sample. I agree it's weird, but appears to be legitimately accepted by ruby.

**Comment by whitequark** [comment]
This was fixed in e188015.
